test/studies/bigint/bigintops.chpl

```diff
@@ -0,0 +1,88 @@
+use BigInteger, Time, List;
+
+config param perfTest = true;
+
+config const numOps = 10;
+
+proc main() {
+  var inputSizes: [0..3] bigint;
+
+  // 64 bit
+  inputSizes[0] = 1:bigint << 63;
+
+  // 128 bit
+  inputSizes[1] = 1:bigint << 127;
+
+  // 256 bit
+  inputSizes[2] = 1:bigint << 255;
+
+  // 1024 bit
+  inputSizes[3] = 1:bigint << 1023;
+
+  for inputSize in inputSizes do
+    runOpTest(inputSize);
+  for inputSize in inputSizes do
+    runAssignTest(inputSize);
+  for inputSize in inputSizes do
+    runOnStmtTest(inputSize);
+}
+
+// Case 1: operator
+proc runOpTest(inputVal: bigint) {
+  var a: bigint = inputVal;
+  var b: bigint = inputVal;
+  var c: bigint;
+
+  var t: stopwatch;
+  if perfTest then t.start();
+  
+  for i in 1..numOps do
+    c = a + b;
+
+  if perfTest {
+    t.stop();
+    writeln("Elapsed ", inputVal.sizeInBase(2), " op time = ", t.elapsed(TimeUnits.seconds));
+  } else {
+    writeln("Result: ", c);
+  }
+}
+
+// Case 2: assignment operator
+proc runAssignTest(inputVal: bigint) {
+  var a: bigint = inputVal;
+  var b: bigint;
+  
+  var t: stopwatch;
+  if perfTest then t.start();
+  
+  for i in 1..numOps do
+    b += a;
+
+  if perfTest {
+    t.stop();
+    writeln("Elapsed ", inputVal.sizeInBase(2), " assign time = ", t.elapsed(TimeUnits.seconds));
+  } else {
+    writeln("Result: ", b);
+  }
+}
+
+// Case 3: operate on self
+proc runOnStmtTest(inputVal: bigint) {
```

Comment:
being pedantic but, I don't know why it refers to an `OnStmt`.

Reply:
Good point. That has something to do with bigint internals, but that isn't particularly useful in the context of this benchmark. Updated that comment.